src/gt4py/next/type_system_2/traits.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+import abc
+from typing import Any, Optional
+from collections.abc import Sequence
+
+import dataclasses
+
+
+class Trait:
+    ...
+
+
+class SignednessTrait(Trait):
+    @abc.abstractmethod
+    def is_signed(self) -> bool:
+        ...
+
+
+class FromTrait(Trait):
+    @abc.abstractmethod
+    def is_constructible_from(self, ty: Any) -> bool:
```

Comment:
```suggestion
    def is_constructible_from(self, ty: types.Type) -> bool:
```
probably, but you might have done this later anyway

Reply:
I don't really know how to avoid the circular import with `types.py` and `traits.py`, but yes, it should definitely not be `Any`.